edward/data.py

```diff
@@ -30,57 +38,71 @@ def __init__(self, data=None, shuffled=True):
             # shuffle self.data
             raise NotImplementedError()
 
-        self.counter = 0
         if self.data is None:
-            self.N = None
+            pass
         elif isinstance(self.data, tf.Tensor):
             self.N = self.data.get_shape()[0].value
+            self.counter = 0
         elif isinstance(self.data, np.ndarray):
             self.N = self.data.shape[0]
+            self.counter = 0
+        elif isinstance(self.data, list):
+            if isinstance(self.data[0], np.ndarray):
+                self.N = [x.shape[0] for x in self.data]
+                self.counter = [0]*len(self.data)
+            else: # list of placeholders
+                # need data set size to scale gradients appropriately
+                pass
         elif isinstance(self.data, dict):
             pass
         else:
-            raise
+            raise NotImplementedError()
 
     def sample(self, n_data=None):
         # TODO scale gradient and printed loss by self.N / self.n_data
-        if n_data is None:
+        if n_data is None or self.data is None:
```

Comment:
the current experiments do the wrong thing because it adds log(scale factor) rather than scale factor * log_likelihood_on_minibatch + log_prior